### PR TITLE
Sample hypersphere for vMF

### PR DESCRIPTION
**Pull request by stas97**
Created a function for sampling a hyper-sphere uniformly. 
This is used during the sampling of vMF. 

@tillaczel : I think we should use this one to create synthetic hypersphere dataset as well. 
Maybe I am wrong, but in the `synthetic_hypersphere.py` from #1: 

As I see it, we generate (x,y,z) where each of the coordinates is uniform(0,1) and then we normalize them to lie on the sphere.  
This is a distribution on [0,1)³ which is transformed to a sphere, whereas my method is actually a distribution on a m-dimensional sphere. 

Anyway, for the purpose of having a synthetic sphere, I do not think this is a huge deal. But since I need this function anyway, we can use it to simulate as well.

### CODE REVIEW

File: s_vae/models/backbone/utils.py [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
         super().__init__()
         self.shape = shape
 
     def forward(self, x):
-        return x.view(-1, *self.shape)
\ No newline at end of file
+        return x.view(-1, *self.shape)
+
+
+def sample_hypersphere(n, dim =3, R = 1, seed = 0):
```
Comment (tillaczel) [inline s_vae/models/backbone/utils.py:12]: I would not put this under backbone as this will not be used by the backbones. Not sure if we will use it for  synthetic data generation, we can move it later as well (as you mentioned). We can create a utils file in `s_vae`?

File: s_vae/models/backbone/utils.py [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
+    """ 
+    This function will sample from uniform distribution on the *dim*-dimensional unit sphere. 
+    The method used comes from  "Computer Generation of Distributions on
+    the m-sphere" By GARY ULRICH. 
+    n: number of points to sample
+    dim: the dimension of the sphere.
+    R: radius of the sphere
+    """
+    torch.seed(seed)
```
Comment (tillaczel) [inline s_vae/models/backbone/utils.py:21]: Seed is set for `torch` by `pytorch lightning`, so this should either be deleted or:
 
```suggestion
def sample_hypersphere(n, dim=3, R=1, seed=None):
    if seed is not None:
        torch.seed(seed)
```

File: s_vae/models/backbone/utils.py [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+    
+    distrib = torch.distributions.MultivariateNormal(torch.zeros(dim), torch.eye(dim))
+    dim_arr = torch.empty(dim, n)
```
Comment (tillaczel) [inline s_vae/models/backbone/utils.py:24]: Data matrix is usually `n` by `dim`. I would use `torch.empty(n, dim)`

File: s_vae/models/backbone/utils.py [hunk 4]
```diff
@@ -1,10 +1,31 @@
...
+
+    for i in range(n):
+        values = distrib.sample()
+        assert(R>0)
+        dim_arr[:,i] = (values*R)/torch.linalg.norm(values, ord= 2)
```
Comment (tillaczel) [inline s_vae/models/backbone/utils.py:29]: This generates a uniform dist on a sphere with radius `R`?

**Review by tillaczel — CHANGES_REQUESTED** [review]
Great work! I would do some minor changes.
